File: ABC/abc014/abc014_3 3.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // 100点解法
    int n, a, b;
    cin >> n;

    vector<pair<int, int>> query;
    for (int i = 0; i < n; i++) {
        cin >> a >> b;
        query.push_back({a, 1});
        query.push_back({b+1, -1});
    }
    sort(query.begin(), query.end());

    // イベントソート(?)
    int num = 0; // 人気
    map<int, int> s;
    for (auto [color, change] : query) {
        num += change;
        s[color] = num;
    }

    int ans = -1;
    for (auto [color, num] : s) {
        if (num > ans) ans = num;
    }

    cout << ans << endl;
}

File: ABC/abc014/abc014_3 3.py
#100点解法
from collections import defaultdict

n = int(input())
query = []
for _ in range(n):
    a, b = map(int, input().split())
    query.append((a, 1))
    query.append((b+1, -1))
query.sort()

#イベントソート(?)
num = 0 #人気
s = defaultdict(int)
for color, change in query:
    num += change
    s[color] = num

print(max(s.values()))